packages/protocol/thread/thread.ts

```diff
@@ -586,7 +592,8 @@ class _ThreadFront {
     const pause = this.pauseForAsyncIndex(asyncIndex);
     assert(pause, "no pause for asyncIndex");
 
-    const rv = await pause.evaluate(frameId, text, pure);
+    const abilities = await this.recordingCapabilitiesWaiter.promise;
+    const rv = await pause.evaluate(frameId, text, abilities.supportsPureEvaluation && pure);
```

Comment:
🛑 FYI the `evaluate()` method is dead as of earlier today, and should be deleted in a follow-up PR - try applying this change to `evaluateNew()` instead

Reply:
Added to the new one as well! Thanks!